core-skills/c-sharp-and-dotnet/README.md

```diff
@@ -0,0 +1,78 @@
+# C# and .NET Core
+
+## Who is this skill for?
+
+This skill is intended for developers who have already completed the TDD core skill, and who have not worked with .NET previously (or want to brush up) and are likely to join a delivery team that is working in C#.
+Through this course, we'll be making some assumptions that you're already familiar with core Made Tech processes and at least one programming language (static typed or dynamic).
+
+## Summary
+### Mantis Shrimp
+1. Can set up a working project in .NET using an IDE
+2. Can install and manage dependencies for the project
+3. Can create and run unit tests using a test framework
+4. Understands basic concepts of the C# language and ecosystem
+
+### Goat
+1. Understands some deeper object oriented programming concepts
+2. Able to demonstrate common core patterns implemented using C# language features
+3. Understands how to build API functionality using common frameworks (MVC, EF)
+4. Understands how to use the dotnet command line utilities to perform common tasks
+5. Can effectively debug using the interactive debugger (IDE)
+
+### Eagle
+1. Demonstrate an understanding of functional programming in C#
+2. Understanding of the .NET virtual machine
+3. Understanding the impact of choices on performance
+4. Demonstrate ability to combine understanding of the above into a working API in C# with .NET
+
+## Resources
+Coming soon
+
+## Mark Scheme
+### Mantis Shrimp
+
+- Can explain the difference between .NET, .NET Core and Framework
+- Can explain the difference between C#, Visual Basic and F#
+- Can explain what assemblies and artefacts are
+- Can explain the concept of classes with examples of usage
+- Can explain the concept of interfaces with examples of usage
+- Can explain the concept of inheritance with examples of usage
+
+- Can create a new project in an IDE
+- Can install and manage external project dependancies
+- Can build and run the project
+- Can manage internal project dependancies
+- Can create a test project and a suite of tests that test some functionality
+- Demonstrates use of classes, interfaces and inheritance
+- Demonstrates the use of scope/access modifiers
+
+Expected deliverable - simple application with a test that satisfies the above
+
+### Goat
+
+- Can explain the concept of generics with examples of usage
+- Can explain the concept of abstraction with examples of usage
+- Can explain the concept of polymorphism with examples of usage
+
+- Can demonstrate understanding and implementation of exception handling
+- Can demonstrate understanding and implementation of asynchronous programming
+- Can demonstrate understanding and implementation of LINQ query syntax and method syntax
```

Comment:
common pitfalls of LINQ including IQueryable, IEnumable and the various liskov substitution violations that occur in the common Framework APIs is useful to cover